python/graphstorm/dataloading/sampler.py

```diff
@@ -70,6 +72,190 @@ def _generate(self, g, eids, canonical_etype):
         dst = F.randint(shape, dtype, ctx, 0, self._local_neg_nids[vtype].shape[0])
         return src, self._local_neg_nids[vtype][dst]
 
+class GSHardEdgeDstNegative(object):
```

Comment:
i think we should name this as hard negative sampler. it should be custom negative edge sampler.